ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -126,18 +126,21 @@
       VALID_WINDOW_BOUNDS_TYPES.toString();
 
   private final KsqlExecutionContext executionContext;
-  private final Optional<HeartbeatAgent> heartbeatAgent;
   private final RoutingFilterFactory routingFilterFactory;
+  private final Optional<PullQueryExecutorMetrics> pullQueryExecutorMetrics;
+  private final Time time;
 
   public PullQueryExecutor(
       final KsqlExecutionContext executionContext,
-      final Optional<HeartbeatAgent> heartbeatAgent,
-      final RoutingFilterFactory routingFilterFactory
+      final RoutingFilterFactory routingFilterFactory,
+      final Optional<PullQueryExecutorMetrics> pullQueryExecutorMetrics
```

Comment:
so, I think we should construct this internally within `PullQueryExecutor` and not pass this from outside.. The goal is to keep the metrics contained to this part of the code, the rest should not have to know about this?